lib/ansible/modules/cloud/amazon/aws_s3.py

```diff
@@ -523,21 +533,22 @@ def download_s3file(module, s3, bucket, obj, dest, retries, version=None):
             key = s3.get_object(Bucket=bucket, Key=obj)
     except botocore.exceptions.ClientError as e:
         if e.response['Error']['Code'] != "404":
-            module.fail_json(msg="Could not find the key %s." % obj, exception=traceback.format_exc(), **camel_dict_to_snake_dict(e.response))
-
+            module.fail_json_aws(e, msg="Could not find the key %s." % obj)
+    except botocore.exceptions.BotoCoreError as e:
```

Comment:
Pre-existing, but above this line I think an exception that is not 404 will just be ignored. Should the error messages be the same not found for exceptions other than 404? I think it would indicate something else was wrong.

Reply:
Not sure I understand what you mean. The code seems to cover three cases:

* Error Code is 404 - object doesn't exist, carry on
* Error Code is 403 - ignore (as it might just be permissions on HeadObject)
* Error Code anything else - fail

Reply:
Sorry, I was just confused and read that wrong. But it looks like a 403 will fail?

Reply:
(which I think is right here, since the user is trying to download it and should be informed if they can't)